controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/CommitRequestHandler.java

```diff
@@ -145,7 +145,7 @@ public CommitRequestHandler(final StreamMetadataStore streamMetadataStore,
                                     CompletableFuture<Void> future;
                                     // if state is sealing, we should continue with commit so that we allow for completion of transactions
                                     // in commit state.
-                                    if (state.equals(State.SEALING)) {
+                                    if (state.getObject().equals(State.SEALING)) {
```

Comment:
This looks like it would change the behavior. Is this a bug that was caught?

Reply:
Yes, this is a bug found by spotbugs, and apparently this condition is not covered by any test.
I am still new to this code, so I did not add a test case, but the bug is clear to me just by looking at the code

Reply:
this is a good catch. 
added issue to add unit test for this scenario https://github.com/pravega/pravega/issues/3122